src/Microsoft.Health.Fhir.SqlServer/Features/Search/Expressions/Visitors/QueryGenerators/SqlQueryGenerator.cs

```diff
@@ -811,4 +838,41 @@ private DenormalizedExpressionContainsIdParameterVisitor()
             public override bool VisitSearchParameter(SearchParameterExpression expression, object context) => expression.Parameter.Name == SearchParameterNames.Id;
         }
     }
+
+    // TODO: move to healthcare-shared-components
+    [SuppressMessage("StyleCop.CSharp.MaintainabilityRules", "SA1402:File may only contain a single type", Justification = "To be moved to healthcare-shared-components")]
+    internal static class IndentedStringBuilderExtensions
+    {
+        /// <summary>
+        /// Helps with building a nested clause with 0 to many predicates joined (ANDed or ORed) together.
+        /// Call <see cref="IndentedStringBuilder.DelimitedScope.BeginDelimitedElement"/> before appending
+        /// a predicate and be sure to dispose the the <see cref="IndentedStringBuilder.DelimitedScope"/>
+        /// at the end.
+        /// </summary>
+        /// <param name="indentedStringBuilder">The string builder</param>
+        /// <param name="delimiter">Delimiter to use for joining. Typically: "AND " or "OR "</param>
+        /// <returns>The scope</returns>
+        public static IndentedStringBuilder.DelimitedScope BeginDelimitedClause(this IndentedStringBuilder indentedStringBuilder, string delimiter)
+        {
+            IndentedStringBuilder.IndentedScope? sharedScope = null;
+            return indentedStringBuilder.BeginDelimitedScope(
+                sb =>
+                {
+                    sb.AppendLine();
+                    sb.AppendLine("(");
+                    sharedScope = sb.Indent();
+                },
+                sb =>
+                {
+                    sb.AppendLine();
+                    sb.Append(delimiter);
+                },
+                sb =>
+                {
+                    sb.AppendLine();
+                    sharedScope?.Dispose();
+                    sb.Append(")");
+                });
+        }
+    }
```

Comment:
This extension method can be removed after this PR is merged: https://github.com/microsoft/healthcare-shared-components/pull/124